rider-fsharp/protocol/src/kotlin/model/RdFSharpTypeProvidersLoaderModel.kt

```diff
@@ -0,0 +1,373 @@
+        +"IsSealed"
+        +"IsAbstract"
+        +"IsPublic"
+        +"IsNestedPublic"
+        +"IsSuppressRelocate"
+        +"IsErased"
+        +"IsGenericType"
+        +"IsMeasure"
+    }
+
+    private val RdProvidedFieldFlags = flags {
+        +"None"
+        +"IsInitOnly"
+        +"IsStatic"
+        +"IsSpecialName"
+        +"IsLiteral"
+        +"IsPublic"
+        +"IsFamily"
+        +"IsFamilyAndAssembly"
+        +"IsFamilyOrAssembly"
+        +"IsPrivate"
+    }
+
+    private val RdProvidedMethodFlags = flags {
+        +"None"
+        +"IsGenericMethod"
+        +"IsStatic"
+        +"IsFamily"
+        +"IsFamilyAndAssembly"
+        +"IsFamilyOrAssembly"
+        +"IsVirtual"
+        +"IsFinal"
+        +"IsPublic"
+        +"IsAbstract"
+        +"IsHideBySig"
+        +"IsConstructor"
+    }
+
+    private val RdTypeName = enum {
+        +"sbyte"
+        +"short"
+        +"int"
+        +"long"
+        +"byte"
+        +"ushort"
+        +"uint"
+        +"ulong"
+        +"decimal"
+        +"float"
+        +"double"
+        +"char"
+        +"bool"
+        +"string"
+        +"dbnull"
+    }
+
+    private val RdProvidedEntityType = flags {
+        +"TypeProvider"
+        +"TypeInfo"
+        +"MethodInfo"
+        +"ConstructorInfo"
+        +"FieldInfo"
+        +"PropertyInfo"
+        +"EventInfo"
+    }
+
+    private val RdTestHost = aggregatedef("RdTestHost") {
+        call("RuntimeVersion", void, string.nullable)
+        call("Dump", void, string)
+    }
+
+    init {
+        RdProvidedPropertyInfo = structdef extends RdProvidedMemberInfo {
+            field("DeclaringType", int)
+            field("PropertyType", int)
+            field("GetMethod", int)
+            field("SetMethod", int)
+            field("CanRead", bool)
+            field("CanWrite", bool)
+            field("IndexParameters", array(RdProvidedParameterInfo))
+        }
+
+        RdProvidedParameterInfo = structdef {
+            field("Name", string.nullable)
+            field("IsIn", bool)
+            field("IsOut", bool)
+            field("IsOptional", bool)
+            field("ParameterType", int)
+            field("RawDefaultValue", RdStaticArg.nullable)
+            field("HasDefaultValue", bool)
+            field("CustomAttributes", array(RdCustomAttributeData).nullable)
+        }
+
+        RdProvidedMethodInfo = structdef extends RdProvidedMethodBase {
+            field("ReturnType", int)
+            field("MetadataToken", int)
+        }
+
+        field("RdTypeProviderProcessModel", RdTypeProviderProcessModel)
+        field("RdProvidedTypeProcessModel", RdProvidedTypeProcessModel)
+        field("RdProvidedMethodInfoProcessModel", RdProvidedMethodInfoProcessModel)
+        field("RdProvidedConstructorInfoProcessModel", RdProvidedConstructorInfoProcessModel)
+        field("RdProvidedAssemblyProcessModel", RdProvidedAssemblyProcessModel)
+        field("RdTestHost", RdTestHost)
+    }
+}

```

Comment:
_(Minor)_ Some of the field names in this file start from uppercase letter, and some from a lowercase one. It would be nice to have single standard (since the generator will fine-tune it for target languages anyway).